pandas/core/indexing.py

```diff
@@ -1816,13 +1821,13 @@ def _setitem_single_block(self, indexer, value):
 
             indexer = maybe_convert_ix(*indexer)
 
-        if isinstance(value, (ABCSeries, dict)):
+        if isinstance(value, (ABCSeries, dict)) and name != "iloc":
```

Comment:
if we have a dict here dont we still want to wrap it in a Series?

Reply:
Yep, have no tests getting here with iloc. Will add one

Reply:
LMK when these tests are all ready and ill take another look. this one is almost ready

Reply:
Tests are all in